drizzle-orm/src/monodriver.ts

```diff
@@ -0,0 +1,387 @@
+} from 'postgres';
+import type { AwsDataApiPgDatabase, DrizzleAwsDataApiPgConfig } from './aws-data-api/pg/index.ts';
+import type { BetterSQLite3Database as DrizzleBetterSQLite3Database } from './better-sqlite3/index.ts';
+import type { BunSQLiteDatabase } from './bun-sqlite/index.ts';
+import type { DrizzleD1Database } from './d1/index.ts';
+import type { LibSQLDatabase } from './libsql/index.ts';
+import type { MySql2Database, MySql2DrizzleConfig } from './mysql2/index.ts';
+import type { NeonHttpDatabase } from './neon-http/index.ts';
+import type { NeonDatabase } from './neon-serverless/index.ts';
+import type { NodePgDatabase } from './node-postgres/index.ts';
+import type { PlanetScaleDatabase } from './planetscale-serverless/index.ts';
+import type { PostgresJsDatabase } from './postgres-js/index.ts';
+import type { TiDBServerlessDatabase } from './tidb-serverless/index.ts';
+import type { DrizzleConfig } from './utils.ts';
+import type { VercelPgDatabase } from './vercel-postgres/index.ts';
+
+type BunSqliteDatabaseOptions =
+	| number
+	| {
+		/**
+		 * Open the database as read-only (no write operations, no create).
+		 *
+		 * Equivalent to {@link constants.SQLITE_OPEN_READONLY}
+		 */
+		readonly?: boolean;
+		/**
+		 * Allow creating a new database
+		 *
+		 * Equivalent to {@link constants.SQLITE_OPEN_CREATE}
+		 */
+		create?: boolean;
+		/**
+		 * Open the database as read-write
+		 *
+		 * Equivalent to {@link constants.SQLITE_OPEN_READWRITE}
+		 */
+		readwrite?: boolean;
+	};
+
+type BunSqliteDatabaseConfig =
+	| {
+		filename?: ':memory:' | (string & {});
+		options?: BunSqliteDatabaseOptions;
+	}
+	| ':memory:'
+	| (string & {})
+	| undefined;
+
+type BetterSQLite3DatabaseConfig =
+	| {
+		filename?:
+			| ':memory:'
+			| (string & {})
+			| Buffer;
+		options?: BetterSQLite3Options;
+	}
+	| ':memory:'
+	| (string & {})
+	| undefined;
+
+type MonodriverNeonHttpConfig = {
+	connectionString: string;
+	options?: NeonHttpConfig<boolean, boolean>;
+};
+
+type VercelPrimitive = string | number | boolean | undefined | null;
+
+type DatabaseClient =
+	| 'node-postgres'
+	| 'postgres-js'
+	| 'neon-serverless'
+	| 'neon-http'
+	| 'vercel-postgres'
+	| 'aws-data-api-pg'
+	| 'planetscale'
+	| 'mysql2'
+	| 'tidb-serverless'
+	| 'libsql'
+	| 'd1'
+	| 'bun:sqlite'
+	| 'better-sqlite3';
+
+type ClientDrizzleInstanceMap<TSchema extends Record<string, any>> = {
+	'node-postgres': NodePgDatabase<TSchema>;
+	'postgres-js': PostgresJsDatabase<TSchema>;
+	'neon-serverless': NeonDatabase<TSchema>;
+	'neon-http': NeonHttpDatabase<TSchema>;
+	'vercel-postgres': VercelPgDatabase<TSchema>;
+	'aws-data-api-pg': AwsDataApiPgDatabase<TSchema>;
+	planetscale: PlanetScaleDatabase<TSchema>;
+	mysql2: MySql2Database<TSchema>;
+	'tidb-serverless': TiDBServerlessDatabase<TSchema>;
+	libsql: LibSQLDatabase<TSchema>;
+	d1: DrizzleD1Database<TSchema>;
+	'bun:sqlite': BunSQLiteDatabase<TSchema>;
+	'better-sqlite3': DrizzleBetterSQLite3Database<TSchema>;
+};
```

Comment:
I'm guessing we'll need to add these types the SingleStore related things

Reply:
I don't think that is necessary because these all look like external packages Drizzle supports, not the dialects. There is no mention of "MySQL", "PostgreSQL" or "SQLite" in this file, only Turso, Bun, mysql2, etc...